### PR TITLE
[feat] Add a `--list-tags` command-line action to list all the available test tags

### LINKED ISSUE #1851: List all available tags of tests
This is a feature request from Cerlane.

### PR DESCRIPTION
**Pull request by teojgo**
Closes #1851
Signed-off-by: Theofilos Manitaras <manitaras@cscs.ch>

### CODE REVIEW

File: reframe/frontend/cli.py [hunk 1]
```diff
@@ -269,6 +277,10 @@ def main():
         '-L', '--list-detailed', action='store_true',
         help='List the selected checks providing details for each test'
     )
+    action_options.add_argument(
+        '--list-tags', action='store_true',
+        help='List the unique tags found in the selected tests'
```
Comment (vkarak) [inline reframe/frontend/cli.py:282]:
```suggestion
        help='List the unique tags found in the selected tests and exit'
```

File: unittests/test_cli.py [hunk 2]
```diff
@@ -499,7 +501,7 @@ def test_timestamp_option(run_reframe):
 
 def test_list_empty_prgenvs_check_and_options(run_reframe):
     returncode, stdout, _ = run_reframe(
-        checkpath=['unittests/resources/checks/frontend_checks.py'],
+        checkpath=['unittests/resources/checks/hellocheck.py'],
```
Comment (vkarak) [inline unittests/test_cli.py:504]: I don't understand this very much...

File: reframe/frontend/cli.py [hunk 3]
```diff
@@ -803,6 +815,10 @@ def _case_failed(t):
             list_checks(testcases, printer, options.list_detailed)
             sys.exit(0)
 
+        if options.list_tags:
+            list_tags(testcases, printer)
+            sys.exit(0)
+
```
Comment (vkarak) [inline reframe/frontend/cli.py:821]: I think that you should add this action later on as well when we list them in case no action is specified.

**Review by vkarak — CHANGES_REQUESTED** [review]
Lgtm, I just have two minor comments.

File: unittests/test_cli.py (hunk 2 above)
Comment (teojgo) [reply unittests/test_cli.py:504]: This happened by mistake I will fix.

**Review by vkarak — APPROVED** [review]

**Pull request merged**